test/duration/parse.test.js

```diff
@@ -74,3 +74,36 @@ test("Duration.fromISO rejects junk", () => {
   rejects("P34K");
   rejects("P5D2W");
 });
+
+//------
+// #fromISOTime()
+//------
+
+const checkTime = (s, ob) => {
+  expect(Duration.fromISOTime(s).toObject()).toEqual(ob);
+};
+
+test("Duration.fromISOTime can parse a variety of extended ISO time formats", () => {
+  checkTime("11:22:33.444", { hours: 11, minutes: 22, seconds: 33, milliseconds: 444 });
+  checkTime("11:22:33", { hours: 11, minutes: 22, seconds: 33 });
+  checkTime("11:22", { hours: 11, minutes: 22, seconds: 0 });
+  checkTime("T11:22", { hours: 11, minutes: 22, seconds: 0 });
+});
+
+test("Duration.fromISOTime can parse a variety of basic ISO time formats", () => {
+  checkTime("112233.444", { hours: 11, minutes: 22, seconds: 33, milliseconds: 444 });
+  checkTime("112233", { hours: 11, minutes: 22, seconds: 33 });
+  checkTime("1122", { hours: 11, minutes: 22, seconds: 0 });
+  checkTime("11", { hours: 11, minutes: 0, seconds: 0 });
+  checkTime("T1122", { hours: 11, minutes: 22, seconds: 0 });
+});
+
```

Comment:
This exactly matches how the time component of a date is parsed - including the, in my opinion, strange behavior where milliseconds are undefined if not specified, while the other values default to 0.

A general discussion could be had about whether defaulting to 0 is the correct behavior, as it means we actually loose the information about the _precision_ of the value. That said, the need for that information is probably an uncommon edge case, and defaulting to 0 is very convenient, as it means users won't need null checks - so I'm ok with that.

However, I do think milliseconds should then also default to 0, both here and when parsing a date, as the current behavior is quite unexpected - especially since the typings for DateTime claim the milliseconds will always be a number...

Thoughts?